examples/host/simpleResolver/simpleResolver.py

```diff
@@ -80,6 +80,25 @@ def print_traits_data(_, data):
     TraitsData. As we only deal with a single result, we ignore the
     index.
     """
+    # Note that in real tools, you should avoid working directly with
+    # TraitsData. The industry-specific view classes should be used to
+    # avoid fragile string literals, and ensure consistency.
+    #
+    # For example, the OpenAssetIO-MediaCreation project[1] provides
+    # traits and specifications for common entity types found in
+    # computer graphics. A tool wanting to find the URL for an image
+    # would use the following trait that defines where an entity's
+    # content can be found:
+    #
+    #   from openassetio_mediacreation.traits.content import LocatableContentTrait
+    #   url = LocatableContentTrait(data).getLocation()
+    #
+    # The low-level API is only used here as this simple example script
+    # hast to work with trait ids supplied by users on the command line.
```

Comment:
We used to have a dynamic runtime registry of traits in the old world, now we have traitGen it would be fairly straight forward to add here. We were looking at this when we switched to Traits but worked out that it only really seemed useful for programatic introspection/UI generation so left it till later as we didn't have a use case - all real-world workflows we modeled (outside of contrived UI) ended up needing to use concrete properties as they had to have first-class knowledge of which trait they needed in order to use the data for something practical. Any generic ones were happy with the low-level TraitsData API.  The only real case we found was a test UI that built a UI for user experimentation.